ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -1872,3 +1873,57 @@ def run_command_as_user(host, user, command, *args, **kwargs):
 
 def kinit_as_user(host, user, password):
     host.run_command(['kinit', user], stdin_text=password + '\n')
+
+
+class FileBackup:
+    """Create file backup and do restore on remote host
+
+    FileBackup(host, filename) creates
+    Also can be used as a context manager.
+    Examples:
+
+    config_backup = FileBackup(host, '/etc/some.conf')
+    ... modify the file and do the test ...
+    config_backup.restore()
+
+    with FileBackup(host, '/etc/some.conf') as config_backup:
+        ... modify the file and do the test ...
+    """
+
+    def __init__(self, host, filename):
+        """Create file backup."""
+        self._host = host
+        self._filename = filename
+        self._backup = create_temp_file(host)
+        host.run_command(['cp', filename, self._backup])
+
+    def restore(self):
+        """Restore file. Can be called multiple times."""
+        self._host.run_command(['cp', self._backup, self._filename])
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.restore()
+
+
+class RemoteIniFile(configparser.RawConfigParser):
+    """Wrap RawConfigParser for modifying .ini files on remote hosts.
```

Comment:
This class does not really inherit the RawConfigParser class. For instance read() and read_file() methods probably won't work unless you override them.
I am wondering whether object composition wouldn't be a better fit here.

Reply:
I replaced this class with a simple context manager.